bilby/gw/waveform_generator.py

```diff
@@ -271,3 +264,288 @@ def validate_reference_frequency(self):
         if SimInspiralGetSpinFreqFromApproximant(waveform_approximant_number) == self.LAL_SIM_INSPIRAL_SPINS_FLOW:
             if self.waveform_arguments["reference_frequency"] != self.waveform_arguments["minimum_frequency"]:
                 raise ValueError(f"For {waveform_approximant}, reference_frequency must equal minimum_frequency")
+
+
+class GWSignalWaveformGenerator(WaveformGenerator):
+    """ A waveform generator with specific checks for GW signals """
```

Comment:
I think this would benefit from a longer doc-string to explain e.g. `spinning` means precessing spins.

Reply:
@ColmTalbot I think this is all that's left.